### PR TITLE
Blog post on GPU course

### PR DESCRIPTION
**Pull request by mondus**

### CODE REVIEW

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+7) Optionally change the UAC control settings to Never Notify if you don't want the user to be prompted about the install when they log on. This of course will have security implications so only do this if you are certain it is not an issue. For my course machines are destroyed after each session so it doesn't matter if users trash the instances.
+
+By following the above, new users on the instance will be able to create a support request from ```https://remotedesktop.google.com/support/``` By clicking the ```Generate Code``` button. This can then be shared with lab demonstrators so that they can share control of the machine which discussing via Google meet.
```
Comment (twinkarma) [inline _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: which -> while?

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+* A description of an instance (image, machine type, VPC, etc.)
+* An availability range (a set of dates in which instances can be created)
+* A fixed time period for the instances (after which any running instance will be terminated)
+* Specification of a list of users (by email address) who are permitted to take the lab and access an instance for a fixed number of sessions.
+	
+Any user who signs up to the site and has been allocated a session on a lab will be able to click the Start Lab button. Behind the scenes the InstanceHub website will start and instance using the Boto3 API and use the User Data approach to create a disposable user account for each user. The USer account will be displayed to users on the website so that they can connect. The above effective allows each student on my GPU course access to a disposable instance a fixed number of times putting a restriction on the spending budget.
```
Comment (twinkarma) [inline _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: USer -> User

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md (hunk 1 above)
Comment (mondus) [reply _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: Resolved in 1e24f861458a9b7cbf9ab3766af422f688ded009

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md (hunk 2 above)
Comment (mondus) [reply _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: Resolved in 1e24f861458a9b7cbf9ab3766af422f688ded009

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+tags: cuda
+category:
+link:
+description:
+type: text
+---
+
+Academics and RSEs have been very busy over the last few weeks coming up with creative solutions to move teaching and training online. My own under/post graduate GPU module [Parallel Computing with GPUs](https://paulrichmond.shef.ac.uk/teaching/COM4521/) has posed a significant problem. The course was designed to be run within the University of Sheffield's High Specification teaching lab which is equipped with CUDA enabled GPUs. Moving this course online clearly requires a mechanism for students to access GPUs, without presenting a significant change to their current working practice (e.g. Visual Studio development in Windows). Ideally this cold be done using the [Universities high performance computing facilities](https://www.sheffield.ac.uk/it-services/research/hpc) but the provision for GPUs is currently insufficient to support 100 students (although new GPUs are on the way). The obvious solution is to move this to the cloud however there are a number of challenges to solve which are the topic of this blog post which also serves as a reference for when I forget all of this in 6 months time.
```
Comment (Robadob) [inline _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: Universities high ->university's high ?

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+*Note: This blog specifically targets AWS as it is what I have used on the [InstanceHub](www.instancehub.com) website which is part of the solution to Problem 3.*
+
+## Problem 1: GPU Accelerated Instances with Graphical Displays
+
+Creating an AWS image with accelerated Rendering and CUDA support is surprisingly more difficult than you might imagine. Having a GPU backed Windows instance still requires a driver install. The are some subtle things to watch out for here. E.g. Most GPU backed based cloud machine types use Tesla GPUs. These will support the NVIDIA CUDA driver but will run as headless devices using [TCC driver mode in Windows](https://docs.nvidia.com/gameworks/content/developertools/desktop/nsight/tesla_compute_cluster.htm). In real speak this means that they will run CUDA applications and support debugging and profiling but won't be involved in driving the display over remote desktop (or VNC). The built in windows display driver will instead be used which has OpenGL 1.0 support only.
```
Comment (Robadob) [inline _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: is "In real speak" really required?

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+For security the lab set up for my GPU course uses a VPC which restricts access to instances to those from within the University of Sheffield. Each student must therefore be on the VPN to access the instance via RDP.
+
+{% include image_caption.html url="/assets/images/instancehub.png" %}
+
+
+## What Next
+
+All of the content described above was to provide an environment which closely matched what was previously available to students on campus. Personally I would prefer to provide Linux images and reproduce much of the above. Linux is less fussy about GPU drivers but more awkward to configure a visual desktop environment with VLC. We (the RSE team) have begun looking at this. Perhaps another blog post will follow at some point.
```
Comment (Robadob) [inline _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: VLC -> VNC?

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md (hunk 3 above)
Comment (mondus) [reply _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: fixed in 1e24f861458a9b7cbf9ab3766af422f688ded009

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md (hunk 4 above)
Comment (mondus) [reply _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: fixed in 1e24f861458a9b7cbf9ab3766af422f688ded009

File: _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md (hunk 5 above)
Comment (mondus) [reply _posts/2020-03-24-configuring-cuda-and-opengl-courses-in-the-cloud.md]: fixed in 1e24f861458a9b7cbf9ab3766af422f688ded009

**Pull request merged**